### PR TITLE
Trusted Identity Types

### PR DESCRIPTION
**Pull request by dolanbernard**
Replace TrustedMrEnclaveMeasurement and TrustedMrSignerMeasurement with TrustedMrEnclaveIdentity and TrustedMrSignerIndentity.

### Motivation

This PR brings new identity types from mc-attestation-verifier into MobileCoin core to replace old measurement types

### CODE REVIEW

File: attest/verifier/config/Cargo.toml [hunk 1]
```diff
@@ -9,8 +9,10 @@ readme = "README.md"
 [dependencies]
 displaydoc = { version = "0.2", default-features = false }
 hex = { version = "0.4.3", default-features = false, features = ["alloc", "serde"] }
+mc-attestation-verifier = "0.2.0"
+mc-sgx-core-types = "0.7.2"
 serde = { version = "1.0", default-features = false, features = ["derive"] }
-serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
+serde_json = { version = "1.0.103", default-features = false, features = ["alloc"] }
```
Comment (nick-mobilecoin) [inline attest/verifier/config/Cargo.toml:15]: :question: Do you know what caused this to change to include the patch version? Curious if it failed to build or similar.

I think we for the attestation and sgx repo we need to remove the patch version as it's going to force tighter constraints then necessary.

```suggestion
serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
```

File: attest/verifier/src/status.rs [hunk 2]
```diff
@@ -14,36 +14,55 @@
...
+    TrustedMrSignerIdentity, Verifier,
 };
 use mc_sgx_core_types::{MrEnclave, MrSigner};
 use mc_sgx_css::Signature;
 use serde::{Deserialize, Serialize};
 
 /// A helper function used to check exceptions to the quote error = fail rule.
-fn check_ids(quote_status: &IasQuoteResult, config_ids: &[String], sw_ids: &[String]) -> bool {
+fn check_ids(quote_status: &IasQuoteResult, advisories: &Advisories) -> bool {
+    let verifier = AdvisoriesVerifier::new(advisories.to_owned()); // TODO: this?
```
Comment (nick-mobilecoin) [inline attest/verifier/src/status.rs:32]: :wrench: Does this `TODO` still belong here?
```suggestion
    let verifier = AdvisoriesVerifier::new(advisories.to_owned());
```

File: attest/verifier/src/status.rs [hunk 3]
```diff
@@ -55,43 +74,17 @@ pub enum Kind {
...
-                Kind::Signer(v) => {
-                    v.allow_config_advisory(id);
-                }
-            }
-        }
-        self
-    }
-
     /// Assume an enclave with the specified measurement has the appropriate
     /// software/build-time hardening for the given advisory IDs.
```
Comment (nick-mobilecoin) [inline attest/verifier/src/status.rs:78]: :wrench: Seems like this comment should be updated, the "software/build-time hardening" seems to imply `SwHardening`
```suggestion
    /// Advisories an enclave is allowed to have.
```

File: attest/verifier/src/status.rs [hunk 4]
```diff
@@ -244,50 +195,13 @@ impl MrSignerVerifier {
...
-    /// Assume an enclave with the specified signer, ID, and version has the
-    /// appropriate software/build-time hardening for the given advisory
-    /// IDs.
-    ///
-    /// This method should only be used when advised by an enclave author.
-    pub fn allow_hardening_advisories(&mut self, ids: &[&str]) -> &mut Self {
-        for id in ids {
-            self.sw_ids.push((*id).to_owned());
-        }
+    /// TODO: Doc
```
Comment (nick-mobilecoin) [inline attest/verifier/src/status.rs:202]: :wrench: Similar to `MrEnclaveVerifier` thinking this `TODO` should be done

**Review by nick-mobilecoin — COMMENTED** [review]
Looks good, mainly some stray TODO comments that look like they should be addressed prior to merge.

File: attest/verifier/src/status.rs [hunk 5]
```diff
@@ -426,14 +335,16 @@ mod test {
...
     #[test]
-    fn mrenclave_config_fail() {
+    fn mrenclave_config_and_sw_pass() {
         let verifier = MrEnclaveVerifier {
             mr_enclave: MrEnclave::try_from(MR_ENCLAVE).expect("BUG: invalid test data"),
-            config_ids: vec!["INTEL-SA-00123".to_owned()],
-            sw_ids: vec!["INTEL-SA-00239".to_owned()],
+            advisories: Advisories::new(
+                &vec!["INTEL-SA-00123", "INTEL-SA-00239"],
+                AdvisoryStatus::ConfigurationAndSWHardeningNeeded,
```
[the remark this answers is not in the document]
Comment (dolanbernard) [reply attest/verifier/src/status.rs:346]: This line replaced:
```
config_ids: vec!["INTEL-SA-00123".to_owned()],
sw_ids: vec!["INTEL-SA-00239".to_owned()],
```
which would be config and sw hardening needed

File: attest/verifier/src/status.rs [hunk 6]
```diff
@@ -472,8 +385,10 @@ mod test {
     fn mrenclave_multi_sw_pass() {
         let verifier = MrEnclaveVerifier {
             mr_enclave: MrEnclave::try_from(MR_ENCLAVE).expect("BUG: invalid test data"),
-            config_ids: vec![],
-            sw_ids: vec!["INTEL-SA-00334".to_owned(), "INTEL-SA-00615".to_owned()],
+            advisories: Advisories::new(
+                &vec!["INTEL-SA-00334".to_owned(), "INTEL-SA-00615".to_owned()],
+                AdvisoryStatus::ConfigurationNeeded,
```
[the remark this answers is not in the document]
Comment (dolanbernard) [reply attest/verifier/src/status.rs:390]: Yes, good catch, thank you

**Pull request merged**